### DISCUSSION TITLE
how to validate if the objects with physics are beyond 10 meters from the player?

### DISCUSSION THREAD

**Discussion by vitalspace** [enable3d/enable3d#301] [Help / Questions]
I have an array that contains all the objects, with the player I detect each of these and iterate them with a foreach, to know which one I am colliding with, then there I get all the positions of the objects that are in the array. the question is how valid if the position of the objects is greater than 10 meters to the position of the player in its axes x, y, and z?

**Comment by yandeu** [comment]
You could add a 5m radius sphere to the player as a ghost object.

Similar to https://enable3d.io/examples/enemy-patrols-on-platform.html

**Comment by vitalspace** [comment]
> You could add a 5m radius sphere to the player as a ghost object.
> 
> Similar to https://enable3d.io/examples/enemy-patrols-on-platform.html

Well I'm going to be more specific, I made a map and adding physics lowers performance a lot, so I wanted to validate the distance of objects more than 10 meters from the player and if it is more than 10 meters destroy the physics. so using "ghost object" doesn't help me on this.

**Reply by yandeu** [reply]
In this case, I guess it is enough to get the objects position and calculate the distance manually.

See
- https://threejs.org/docs/#api/en/math/Vector3.distanceTo
- https://threejs.org/docs/#api/en/math/Box3.getCenter

**Comment by vitalspace** [comment]
https://user-images.githubusercontent.com/29004070/175869981-c1dbb7e5-247e-494b-a041-f24c1889b419.mp4

ok, it seems that using ```Vector3.distaceTo```, worked for me.


```
        this.street = this.gltfobject.parent.children;

        this.man.body.on.collision((otherObject, event) => {
          const obx = parseInt(otherObject.position.x);
          const oby = parseInt(otherObject.position.y);
          const obz = parseInt(otherObject.position.z);

          if (otherObject && event === "start") {
            this.street.map((e) => {
              const ex = parseInt(e.position.x);
              const ey = parseInt(e.position.y);
              const ez = parseInt(e.position.z);
              const a = new THREE.Vector3(ex, ey, ez);
              const b = new THREE.Vector3(obx, oby, obz);
              const c = a.distanceTo(b);

              if (e !== otherObject && parseInt(c) > 15) {
                e.visible = false;
              } else {
                e.visible = true;
              }
            });
          }
        });


```
